pkg/selectors/binaryentry.go

```diff
@@ -27,7 +35,21 @@ func (v *BinaryMapValue) NewValue() bpf.MapValue         { return &BinaryMapValu
 func (v *BinaryMapValue) GetValuePtr() unsafe.Pointer    { return unsafe.Pointer(v) }
 func (v *BinaryMapValue) DeepCopyMapValue() bpf.MapValue { return &BinaryMapValue{} }
 
-func writeBinaryMap(id int, path string, m *bpf.Map) error {
+func WriteBinaryMap(id uint32, path string) error {
+	mapDir := bpf.MapPrefixPath()
+	m, err := bpf.OpenMap(filepath.Join(mapDir, base.NamesMap.Name))
```

Comment:
I think using this path means that if two kprobes use the binary  names feature, one will override the map of the other, and things will break. Shouldn't we make this a per-hook map?

There is some code for doing that. Using `filter_map` as an example:

It's added here to the sensor:
https://github.com/cilium/tetragon/blob/3076ae7b9cd2bbd7320af3708af240caa0fc684a/pkg/sensors/tracing/generickprobe.go#L498-L499

Note that the pinPath is set to be hook-specific here:
https://github.com/cilium/tetragon/blob/3076ae7b9cd2bbd7320af3708af240caa0fc684a/pkg/sensors/tracing/generickprobe.go#L476
And it's loaded using `MapLoad` in:
https://github.com/cilium/tetragon/blob/3076ae7b9cd2bbd7320af3708af240caa0fc684a/pkg/sensors/tracing/selectors.go#L59-L67

So the general idea is to store the needed information in the `KernelSelectorState`, and then have the mapLoad callback to load the data when the sensor is loaded.